lib/Common/Memory/RecyclerWriteBarrierManager.cpp

```diff
@@ -66,6 +66,16 @@ X64WriteBarrierCardTableManager::OnThreadInit()
     ::GetCurrentThreadStackLimits(&stackEnd, &stackBase);
 #endif
 
+#ifdef X64_WB_DIAG
+    this->_stackbase = (char*)stackBase;
+    this->_stacklimit = (char*)stackEnd;
+#endif
+
+    // on Windows server 2012 stack limit can expand with process running, and causes
+    // accessing uncommitted card table page.
+    // TODO: use VirtualQuery twice to get the max possible stack limit
```

Comment:
no, just mention here that might be most 'correct' solution. current solution should be enough.

---
In reply to: [89222281](https://github.com/Microsoft/ChakraCore/pull/2071#discussion_r89222281) [](ancestors = 89222281)